mono/metadata/threads.c

```diff
@@ -4866,6 +4969,24 @@ async_abort_internal (MonoInternalThread *thread, gboolean install_async_abort)
 }
 
 static void
+async_abort_internal_handle (MonoInternalThreadHandle thread, gboolean install_async_abort)
+{
+	AbortThreadData data;
+
+	g_assert (MONO_HANDLE_RAW (thread) != mono_thread_internal_current ());
+
+	/* FIXME: Should AbortThreadData.thread be a MonoInternalThreadHandle? */
+	data.thread = MONO_HANDLE_RAW (thread);
+	data.install_async_abort = install_async_abort;
+	data.interrupt_token = NULL;
+
+	mono_thread_info_safe_suspend_and_run (thread_get_tid_handle (thread), TRUE, async_abort_critical, &data);
+	if (data.interrupt_token)
+		mono_thread_info_finish_interrupt (data.interrupt_token);
+	/*FIXME we need to wait for interruption to complete -- figure out how much into interruption we should wait for here*/
+}
```

Comment:
So for situations like this (an existing static function `async_abort_internal` that has multiple callers, some of which have been converted to use handles, and some which haven't), I typically rewrite the existing function (`async_abort_internal`) in terms of `async_abort_internal_handle` and add a `FIXME` to `async_abort_internal`'s new body to remember to rewrite all of its callers.

(Another approach for `MonoInternalThread*` might be to rewrite the `_handle` version in terms of the raw pointer version since all `MonoInternalThread` objects are pinned.)

Reply:
I’ll do that if there’s stuff that needs to depend on the non-handle version. But my idea was to have one commit per function while working on this, then squash away the temporary duplication at the end when everything is converted.

Reply:
Oh, I thought you were going to commit this PR with just the current set of converted functions and then return to the other ones later.